### DISCUSSION TITLE
LESSON 4 Problem error saying"solcx.exceptions.SolcNotInstalled: solc 0.6.0 has not been installed. Use solcx.install_solc('0.6.0')

### DISCUSSION THREAD

**Discussion by chioke07** [smartcontractkit/full-blockchain-solidity-course-py#1852] [Q&A] [answered]
<img width="1512" alt="Screenshot 2023-03-29 at 10 46 14 PM" src="https://user-images.githubusercontent.com/129320402/228639014-aaa743bd-b0d4-41a4-8544-d09768823c64.png">
Hello, I'm taking your course on smart contracts but I got stuck in lesson 4 where you print(compile_sol), anytime I deploy I get an error saying"solcx.exceptions.SolcNotInstalled: solc 0.6.0 has not been installed. Use solcx.install_solc('0.6.0') to install" how can I fix this? I use a mac.

I have used all the information I saw online but still having the same issue, I have installed PIP and also install pip install py-solc-x
.

please help how can I fix this

**Comment by cromewar** [comment, marked as answer]
Hello @chioke07 
The reason is because you have to install the solc version first.
Please refer to: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md#lesson-4 Does this help? let us know.

**Comment by chioke07** [comment]
it working fine now, thanks 
<img width="1512" alt="Screenshot 2023-03-30 at 12 57 23 AM" src="https://user-images.githubusercontent.com/129320402/228665849-4942e496-0857-4e55-a4b9-646c5e3b3f1a.png">

**Reply by cromewar** [reply]
You are welcome!
